### PR TITLE
Publish character set distribution result as output datastream

### LINKED ISSUE #549: OutputRowCollector based on original column
The OutputRowCollector is based on the column headers in the original file, instead of the ones that the component actually want to output. This only becomes a problem if the column count differs between them, as DefaultRow only validates that. It has fallen through the cracks as both tests and currently implemented (proprietary) components has been using the same column count as the input data.

Instead, base the OutputRowCollector on the Table in the OutputStream.

### PR DESCRIPTION
**Pull request by LosD**
Fixes #549

### CODE REVIEW

File: components/i18n/src/main/java/org/datacleaner/beans/CharacterSetDistributionAnalyzer.java [hunk 1]
```diff
@@ -162,6 +171,40 @@ public CharacterSetDistributionResult getResult() {
...
+            }
+        }
+
         return new CharacterSetDistributionResult(_columns, unicodeSetNames, crosstab);
     }
+
+
+    @Override
+    public OutputDataStream[] getOutputDataStreams() {
+        final OutputDataStreamBuilder streamBuilder = OutputDataStreams.pushDataStream("character sets");
```
Comment (tomaszguzialek) [inline components/i18n/src/main/java/org/datacleaner/beans/CharacterSetDistributionAnalyzer.java]: Let's use a constant.

File: components/i18n/src/test/java/org/datacleaner/beans/CharacterSetDistributionAnalyzerTest.java [hunk 2]
```diff
@@ -130,4 +150,76 @@ public void testSimpleScenario() throws Exception {
         assertEquals("Thaana                0      0 ", resultLines[23]);
         assertEquals("Thai                  0      0 ", resultLines[24]);
     }
+
+    public void testOutputDataStream() throws Throwable {
+        final MultiThreadedTaskRunner taskRunner = new MultiThreadedTaskRunner(16);
+        final DataCleanerEnvironment environment = new DataCleanerEnvironmentImpl()
+                .withTaskRunner(taskRunner);
+        final FileResource file = new FileResource("src/test/resources/strings.txt");
```
Comment (tomaszguzialek) [inline components/i18n/src/test/java/org/datacleaner/beans/CharacterSetDistributionAnalyzerTest.java]: I would prefer to have a more informative name for the file. Maybe testOutputDataStream.txt?

**Comment by tomaszguzialek** [comment]
LGTM.

**Comment by kaspersorensen** [comment]
Ehm but isn't the feature here quite strange? I understand we needed an example for testing but this isn't something I would thing anyone wants to have, so it just adds complexity and slightly nonsensical feeling to the user.

**Comment by LosD** [comment]
I somehow agree, although it is possible you'd want to analyze the number distribution of... Errrr, your character sets. 

Patten analyzer and value distribution would probably make more sense, but in the current design, those are not possible because of the multiple analyzers for multiple columns, as the output streams does not get initialized in that case, and we're not entirely sure if that is by design. 

Anyway, we needed _something_ and this one was easy, and is easy to rip out again when something better comes along. And it already served its purpose by uncovering a bug :)

**Comment by kaspersorensen** [comment]
Glad about the bug of course :-) 

The problem is that the moment someone starts using it then we need to have continued support for it and so on. Just removing it would render jobs that use it invalid. I would propose instead two things:

An output stream om Completeness Analyzer called "Incomplete records". And maybe another called "Complete records". At least I have had a few users wish for that, so it's something better worth doing and supporting.

Create another mock implementation in testware and include testware as an extension if you need to test.

**Comment by LosD** [comment]
Perfect idea! I'll get right on that

**Pull request merged**